src/libclient/measurement/extupnp/extupnp.cpp

```diff
@@ -0,0 +1,295 @@
+            {
+                resultHash.insert(NumberOfPortMappings, num);
+            }
+
+            // TODO GetListOfPortMappings do we need this?
+
+            int firewallEnabled, inboundPinholeAllowed;
+
+            if (UPNPCOMMAND_SUCCESS == UPNP_GetFirewallStatus(urls.controlURL,
+                                                              data.first.servicetype,
+                                                              &firewallEnabled,
+                                                              &inboundPinholeAllowed))
+            {
+                resultHash.insert(FirewallEnabled, firewallEnabled);
+                resultHash.insert(InboundPinholeAllowed, inboundPinholeAllowed);
+            }
+
+            int bufferSize = 0;
+            if (char *buffer = (char *)miniwget(urls.rootdescURL, &bufferSize, 0))
+            {
+                NameValueParserData pdata;
+                ParseNameValue(buffer, bufferSize, &pdata);
+                free(buffer);
+                buffer = NULL;
+                QStringList modelName = GetValsFromNameValueList(&pdata, "modelName");
+
+                if (!modelName.isEmpty())
+                {
+                    resultHash.insert(ModelName, modelName.last());
+                }
+
+                QStringList manufacturer = GetValsFromNameValueList(&pdata, "manufacturer");
+
+                if (!manufacturer.isEmpty())
+                {
+                    resultHash.insert(Manufacturer, manufacturer.last());
+                }
+
+                QStringList friendlyName = GetValsFromNameValueList(&pdata, "friendlyName");
+
+                if (!friendlyName.isEmpty())
+                {
+                    resultHash.insert(FriendlyName, friendlyName.last());
+                }
+                qDebug() << friendlyName;// + modelName + manufacturer;
+
+                ClearNameValueList(&pdata);
+            }
+        }
+        /* These URLs will be needed for accessing and controlling Mediaservers with SOAP */
+        QString controlURL = urls.controlURL;
+        if(!controlURL.isEmpty())
+        {
+            resultHash.insert(ControlURL, controlURL);
+        }
+        QString rootDescURL = urls.rootdescURL;
+        if(!rootDescURL.isEmpty())
+        {
+            resultHash.insert(RootDescURL, rootDescURL);
+        }
+        FreeUPNPUrls(&urls);
+        results.append(resultHash);
+        myResults.append(resultHash);
+        list = list->pNext;
+    }
+    freeUPNPDevlist(listBegin);
+    return myResults;
+}
+
+Result ExtUPnP::result() const
+{
+     // List for all results
+    QVariantList deviceResultList;
+
+    foreach (ExtUPnPHash resultHash, results)
+    {
+        QHashIterator<ExtUPnP::DataType, QVariant> iter(resultHash);
+
+        // results from one interface
+        QVariantMap deviceResult;
+
+        while (iter.hasNext())
+        {
+            iter.next();
```

Comment:
I tried it with a for loop, which can also be faound in the code right now.
I commented it and went back because a sigabrt happened everytime i went through the hash.